addons/snailmail/static/src/messaging/component/snailmail_error_dialog/snailmail_error_dialog.js

```diff
@@ -0,0 +1,107 @@
+odoo.define('snailmail.messaging.component.SnailmailErrorDialog', function (require) {
+'use strict';
+
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const Dialog = require('web.OwlDialog');
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+class SnailmailErrorDialog extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const message = this.env.entities.Message.get(props.messageLocalId);
+            const notifications = message ? message.notifications : undefined;
+            return {
+                message: message ? message.__state : undefined,
+                notifications: notifications.map(notification => notification ? notification.__state : undefined),
+                snailmail_credits_url: this.env.messaging.snailmail_credits_url,
+                snailmail_credits_url_trial: this.env.messaging.snailmail_credits_url_trial,
+            };
+        });
```

Comment:
missing compareDepth, and missing guard on notifications before map (or default it to array)

Reply:
done (default to array)